src/main/java/com/neonlab/product/apis/UpdateProductApi.java

```diff
@@ -18,24 +21,32 @@ public class UpdateProductApi {
 
     @Autowired
     private ProductService productService;
-
-    public ApiOutput<ProductDto> updateProduct(ProductDto product) {
+    @Autowired
+    private DocumentService documentService;
+    public ApiOutput<ProductDto> updateProduct(ProductDto product, List<MultipartFile> files, Long documentId) {
 
         try {
-            validate(product);
-            ProductDto productDto = productService.updateProduct(product);
+            validate(product,files);
+            DocumentDto images = documentService.updateDocument(files,documentId);
+            ProductDto productDto = productService.updateProduct(product,images);
             return new ApiOutput<>(HttpStatus.OK.value(),"Product Update Successfully",productDto);
-        }catch (InvalidInputException | ServerException e) {
+        }catch (Exception e) {
             return new ApiOutput<>(HttpStatus.BAD_REQUEST.value(), e.getMessage());
         }
     }
 
-    private void validate(ProductDto productDto) throws InvalidInputException {
+    private void validate(ProductDto productDto, List<MultipartFile> files) throws InvalidInputException {
         if(Objects.isNull(productDto)){
             throw new NullPointerException("Please Add Something To update Product");
         }
         if(StringUtil.isNullOrEmpty(productDto.getCode())){
             throw new InvalidInputException("Product Code is Mandatory");
         }
+        if(files.size()>4){
+            throw new InvalidInputException("Cannot upload more than 4 images.");
+        }
+        if(files.isEmpty()){
+            throw new InvalidInputException("Please Add at-least one file");
+        }
```

Comment:
Remove this validation. User may choose not upload the images because he currently doesn't have the same. Just because he doesn't upload image , we will not allow user to update any field. This is wrong.

cc- @sjha24